client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue

```diff
@@ -108,18 +108,22 @@ const invocationSchedulingTerminal = computed(() => {
     );
 });
 const jobStatesTerminal = computed(() => {
+    // If the job states summary is null, we haven't fetched it yet
+    if (jobStatesSummary.value === null) {
+        return false;
+    }
+
```

Comment:
A proposed solution for the problem pointed out below (the other comment) could be to add this additional check after this ^:
```suggestion

    // For each `step.job_id`, does the `jobCount` match the number of unique `job_id`s?
    const steps = invocation.value?.steps || [];
    const stepJobIds = steps.map((step) => step.job_id).filter((jobId) => jobId !== null);
    const uniqueJobIdCount = new Set(stepJobIds).size;
    if (uniqueJobIdCount > jobCount.value) {
        // not all jobs are accounted for: more job ids than `jobCount` from `jobStatesSummary`
        // (keep polling)
        return false;
    }
```

I had it as `if (uniqueJobIdCount !== jobCount.value)` but there are cases where we have fewer job ids than `jobCount` in the states summary; which made me not do this in the first place... But for more ids than state count, is that still a valid check?

Reply:
Nevermind, I have a solution that might be better